hatch_pip_compile/plugin.py

```diff
@@ -293,16 +294,40 @@ def get_piptools_environment(self, environment_name: str) -> "PipCompileEnvironm
             matrix_variables=self.matrix_variables,
             data_directory=self.data_directory,
             isolated_data_directory=self.isolated_data_directory,
-            platform=self.platform,
+            platform=Platform(),
             verbosity=self.verbosity,
-            app=self.app,
+            app=None,
         )
 
-    def piptools_validate_lock(
+    @property
+    def constraint_env(self) -> "PipCompileEnvironment":
+        """
+        Get the constraint environment
+        """
+        if not self.__constraint_env:
+            constraint_env = self.config.get("pip-compile-constraint")
+            if constraint_env is None:
+                self.__constraint_env = self
+                return self.__constraint_env
+            elif self.name == constraint_env:
+                self.__constraint_env = self
+                return self.__constraint_env
+            environment = self.get_piptools_environment(environment_name=constraint_env)
+            if environment.config.get("type") != self.PLUGIN_NAME:
+                logger.error("The constraint environment is not a hatch-pip-compile environment.")
+                self.__constraint_env = self
+                return self.__constraint_env
+            else:
+                self.__constraint_env = environment
+                return self.__constraint_env
+        else:
+            return self.__constraint_env
```

Comment:
There's a neat utility...
Untested (and wrongly indented because I'm typing on mobile 🤪) but should work

```suggestion
    @functools.cached_property
    def constraint_env(self) -> "PipCompileEnvironment":
        """
        Get the constraint environment
        """
            constraint_env = self.config.get("pip-compile-constraint")
            if constraint_env is None:
                return self
            if self.name == constraint_env:
                return self
            environment = self.get_piptools_environment(environment_name=constraint_env)
            if environment.config.get("type") != self.PLUGIN_NAME:
                logger.error("The constraint environment is not a hatch-pip-compile environment.")
                return self
            else:
                return environment
```

Reply:
And seems that _check_lockfile_up_to_date could make use of cached_property in the same way

Reply:
Neat, TIL `functools.cached_property`. That's super useful and much cleaner, thank you.